### PR TITLE
Migration from old user API to new API

### PR DESCRIPTION
**Pull request by sudheshna-donthineni**
First step in migrating our old backend to the new.
The only main change is that we won't have two different APIs for ID and anything else. 
Instead, we have only one API which will validate whatever student params we send it.

Test Cases:
[DelhiStudents](https://staging-auth.avantifellows.org/?sessionId=DelhiStudents_11B42_44827_ubm-pczd-ntn)
[EnableStudents](https://staging-auth.avantifellows.org/?sessionId=EnableStudents_B02_44801)
[HimachalStudents]()

### CODE REVIEW

File: .github/workflows/deploy_to_s3_prod.yml [hunk 1]
```diff
@@ -15,49 +15,50 @@ jobs:
...
-      uses: aws-actions/configure-aws-credentials@v1
-      with:
-        aws-access-key-id: ${{ secrets.AWS_S3_ACCESS_KEY_ID }}
-        aws-secret-access-key: ${{ secrets.AWS_S3_SECRET_ACCESS_KEY }}
-        aws-region: ap-south-1
-
-    # sync with the given S3 bucket
-    - name: Deploy static site to S3 bucket
-      run: aws s3 sync ./dist s3://af-auth-layer --delete --acl public-read
+      - name: Checkout
```
Comment (pritamps) [inline .github/workflows/deploy_to_s3_prod.yml:18]: Why did you change the indentation? Doesn't seem to be changed for other sections

File: .github/workflows/deploy_to_s3_staging.yml [hunk 2]
```diff
@@ -16,49 +16,50 @@ jobs:
       url: "https://staging-auth.avantifellows.org"
 
     steps:
-    - name: Checkout
-      uses: actions/checkout@v2
+      - name: Checkout
```
Comment (pritamps) [inline .github/workflows/deploy_to_s3_staging.yml:19]: Same question here

File: src/components/Entry.vue [hunk 3]
```diff
@@ -525,14 +525,13 @@ export default {
...
-
       let userValidationResponse = await validateID(
         userID,
-        this.groupData.dataSource,
         this.authType,
         this.validateCount,
         this.dateOfBirth,
-        this.isExtraInputValidationRequired
+        this.isExtraInputValidationRequired,
+        this.$refs.phoneNumberEntry.phoneNumber
```
Comment (pritamps) [inline src/components/Entry.vue:534]: Why is `phoneNumberEntry` coming from `$refs` unlike the others?

File: src/services/API/user.js [hunk 4]
```diff
@@ -1,9 +1,8 @@
-import { client } from "@/services/API/rootClient.js";
+import { client, newClient } from "@/services/API/rootClient.js";
```
Comment (pritamps) [inline src/services/API/user.js]: `newClient` is a very bad name. Call it `AFDBClient` maybe? Or something descriptive

File: src/services/validation.js [hunk 5]
```diff
@@ -88,47 +42,56 @@ async function checkUserIdInFirestore(
...
-      if (authType.includes("OTP")) {
-        return checkPhoneNumberInFirestore(
-          userID,
-          dataSource["name"],
-          dataSource["column"]
-        );
+      if (userValidationResponse) {
+        if (authType.includes("DOB")) {
+          let dob =
+            (birthdate.day < 10 ? "0" + birthdate.day : birthdate.day) +
```
Comment (pritamps) [inline src/services/validation.js]: Are we 100% sure days<10 will be stored **without** the leading 0?

File: src/services/validation.js [hunk 6]
```diff
@@ -88,47 +42,56 @@ async function checkUserIdInFirestore(
...
+            "-" +
+            birthdate.year;
+          var isBirthdateValid = await userAPI.verifyStudent({
+            student_id: userID,
+            birth_date: dob,
+          });
+          return userValidationResponse && isBirthdateValid;
+        }
+        if (authType.includes("PH")) {
+          console.log(authType);
```
Comment (pritamps) [inline src/services/validation.js]: Remove `console.log`

File: src/services/validation.js [hunk 7]
```diff
@@ -88,47 +42,56 @@ async function checkUserIdInFirestore(
...
-          userID,
-          dataSource["name"],
-          dataSource["column"]
-        );
+      if (userValidationResponse) {
+        if (authType.includes("DOB")) {
+          let dob =
+            (birthdate.day < 10 ? "0" + birthdate.day : birthdate.day) +
+            "-" +
+            (birthdate.month < 10 ? "0" + birthdate.month : birthdate.month) +
```
Comment (pritamps) [inline src/services/validation.js:70]: Same question for year.

**Review by pritamps — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy_to_s3_prod.yml (hunk 1 above)
Comment (sudheshna-donthineni) [reply .github/workflows/deploy_to_s3_prod.yml:18]: This seems to have happened with `pre-commit`.

File: src/components/Entry.vue (hunk 3 above)
Comment (sudheshna-donthineni) [reply src/components/Entry.vue:534]: Because `phoneNumberEntry` is a separate component. The ID and DOB are in the same `Entry` component. Moving forward I am creating a separate component for each kind of entry type, so that they can be mixed and matched for different forms.

File: src/services/validation.js (hunk 5 above)
Comment (sudheshna-donthineni) [reply src/services/validation.js]: They are stored with the leading 0. The entry does not have a leading 0, so to compare we append 0 before sending to the backend.

**Review by pritamps — APPROVED** [review]
Looks good to me!

**Review by suryabulusu — COMMENTED** [review]
looks good!
